agent/http_test.go

```diff
@@ -462,15 +463,29 @@ func TestParseConsistency(t *testing.T) {
 	if !b.RequireConsistent {
 		t.Fatalf("Bad: %v", b)
 	}
+
```

Comment:
I think I'd prefer if you refactor the two `ParseInconsistency` tests as follows to ensure that we cover all cases:

```go
func TestParseConsistency(t *testing.T) {
    t.Run("no stale, no consistent, no allowStale", func(t *testing.T) { ... })
    t.Run("stale, no consistent, no allowStale", func(t *testing.T) { ... })
    ... tests for all eight combinations
}
```